src/pages/p_doc.php

```diff
@@ -87,6 +87,11 @@ static function go($user, $qreq) {
             self::error(isset($whyNot["permission"]) ? "403 Forbidden" : "404 Not Found", MessageItem::error("<5>" . $whyNot->unparse_html()), $qreq);
         }
         $prow = $dr->prow;
+
+        if(!$user->privChair && $prow->conf->settings["pc_hideconflicted"] == 1 && $prow->has_conflict($user) && !$row->has_author($pl->user)) {
+            self::error("403 Forbidden", MessageItem::error("<5>Access denied to conflicting submission."), $qreq);
+        }
+
```

Comment:
This should not be needed. The immediately preceding logic should suffice. If you want a special error message you could implement that logic in `src/failurereason.php`.